juspay/hyperswitch
# feat(routing): Integrate global success rates

## Type of Change



## Description

Integrate global Success rates for dynamic routing


### Additional Changes





## Motivation and Context



## How did you test it?

1. toggle success based routing with dynamic connector selection enabled
```
curl --location --request POST 'http://localhost:8080/account/sarthak2/business_profile/pro_F5mC1GGXHPLTtfcBnZEP/dynamic_routing/success_based/toggle?enable=dynamic_connector_selection' \
--header 'api-key: dev_RQxKf8JqHb5tTfTp13JMnI7TyWgXIHaa2AP5VkjLtst1CRJPArSfqasfGQI1oIAQ'
```
2. Set volume split for dynamic routing
```
curl --location --request POST 'http://localhost:8080/account/sarthak2/business_profile/pro_F5mC1GGXHPLTtfcBnZEP/dynamic_routing/set_volume_split?split=80' \
--header 'api-key: dev_RQxKf8JqHb5tTfTp13JMnI7TyWgXIHaa2AP5VkjLtst1CRJPArSfqasfGQI1oIAQ'
```
3. Make a payment
4. Global SR connector should be populated in DB
<img width="639" alt="image" src="https://github.com/user-attachments/assets/1fadcb60-21d1-4958-8511-9a3872bb3daf" />


## Checklist
